apps/prairielearn/src/ee/pages/lti13CourseNavigation/lti13CourseNavigation.html.ts

```diff
@@ -0,0 +1,244 @@
+                              return html`<option value="${ci.id}">
+                                ${course.short_name} -- ${ci.long_name} (${ci.short_name})
+                              </option>`;
+                            })}
+                          </optgroup>
+                        `;
+                      })}
+                    </select>
+                  </div>
+                  <button class="btn btn-primary btn-lg" id="saveButton" disabled>Save</button>
+                </form> `}
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+
+export function Lti13CourseNavigationNotReady({
+  courseName,
+  resLocals,
+}: {
+  courseName: string;
+  resLocals: Record<string, any>;
+}): string {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../../../pages/partials/head')%>", {
+          ...resLocals,
+          navPage: 'lti13_course_naviation',
+          pageTitle: 'Course',
+        })}
+      </head>
+      <body>
+        ${renderEjs(__filename, "<%- include('../../../pages/partials/navbar'); %>", {
+          ...resLocals,
+          navPage: 'lti13_course_navigation',
+        })}
+        <main class="container mb-4">
+          <h1 class="h1">Welcome to PrairieLearn</h1>
+          <h2 class="h2">... but your course isn't ready yet!</h2>
+
+          <p>An instructor has not yet configured ${courseName} in PrairieLearn.</p>
+          <p>Please come back later.</p>
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+
+export function Lti13CourseNavigationDone({
+  resLocals,
+  lti13_instance_id,
+}: {
+  resLocals: Record<string, any>;
+  lti13_instance_id: string;
+}): string {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../../../pages/partials/head')%>", {
+          ...resLocals,
+          navPage: 'lti13_course_naviation',
+          pageTitle: 'Course',
+        })}
+      </head>
+      <body>
+        ${renderEjs(__filename, "<%- include('../../../pages/partials/navbar'); %>", {
+          ...resLocals,
+          navPage: 'lti13_course_navigation',
+        })}
+        <main class="container mb-4">
+          <h1 class="h1">Welcome to PrairieLearn</h1>
+
+          <p>
+            <strong>OK, you're all set.</strong> Next time you (or students) click on the link in
+            your LMS, they will be taken directly to your PrairieLearn course instance.
+          </p>
+          <ul>
+            <li>
+              The course instance <code>allowAccess</code> rules still apply and may need to be
+              configured for the course.
+            </li>
+          </ul>
+
+          <p>
+            To change this connection, go to your course instance "Settings" in the "LTI 1.3" tab.
+          </p>
+
+          <p>
+            <a
+              href="/pl/lti13_instance/${lti13_instance_id}/course_navigation"
```

Comment:
Is there a reason we bounce the user back to the course navigation page instead of directly to their course instance?

Reply:
In development, it was helpful to confirm the settings were actually saved and that the redirect worked. IIRC, it's been a while, but I think all of the things I would do in the Done page to look up and redirect are essentially what a GET to that page would do anyway.

Reply:
Alternatively we could link into the course instance with a flash message? The LTI page has more beginner documentation that wouldn't really fit in a flash message.

Reply:
I have no issues with redirecting folks to the `?done` variant of the course navigation page, I was just curious why the button on _that_ page takes one back to the same page. In testing, I figured out that this currently has the benefit that `ltiClaim.remove();` is called on that page when an `lti13_course_instance` is found. We could probably just as easily call that when loading the `?done` page. That said... calling that on the non-`?done` page has the nice property that trying to navigate back once the link is created will bounce them to the linked course instance instead of possibly allowing them to create another link.

So I might have just rubber-ducked my way into thinking that this is all OK.

Reply:
Yeah, it seemed cleaner to me to do the logic in one route and to keep redirecting the user through it. 

The one place I expect to use `ltiClaim` data outside of the lti13CourseNavigation page is in `instructorRequestCourse` to prepopulate, or prompt to, that request form based on LTI context data. But that can come later (and the claims stay in the session until the course navigation works, so there shouldn't be any other work needed here to allow that.)